src/sentry/static/sentry/app/components/acl/demoModeGate.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import ConfigStore from 'app/stores/configStore';
+import {Organization} from 'app/types';
+import withOrganization from 'app/utils/withOrganization';
+
+type Props = {
+  /**
+   * Current Organization
+   */
+  organization: Organization;
+
+  /**
+   * Children can be a node or a function as child.
+   */
+  children?: React.ReactNode;
+
+  demoComponent?:
+    | React.ReactNode
+    | ((props: {children?: React.ReactNode}) => React.ReactNode);
+};
+
+/**
+ * Component to handle demo mode switches
+ */
+class DemoModeGate extends React.Component<Props> {
+  static defaultProps = {
+    demoComponent: null,
+  };
+
+  render() {
+    const {organization, children, demoComponent} = this.props;
```

Comment:
This could be simplified to just a React function component with:

```
const {organization, children, demoComponent = null} = props;
```

Reply:
@dashed good point! Fixed.